--- 43disable_updateinitramfs	2017-08-02 15:34:13.163476000 +0100
+++ 43disable_updateinitramfs_expected	2017-08-04 09:50:20.442967993 +0100
@@ -24,23 +24,31 @@
 chroot /root dpkg-divert --add --rename --quiet \
         /usr/sbin/update-initramfs
 
-# Running off a USB disk or other writable media.
-if [ ! -e /root/usr/sbin/update-initramfs ] && \
-   [ ! -L /root/usr/sbin/update-initramfs ]; then
-    if [ -w /root/cdrom ] && \
-       # rw is guaranteed to be first.
-       grep -q ' /root/cdrom rw[, ]' /proc/self/mountinfo; then
 
-        ln -s /usr/share/casper/casper-update-initramfs \
-            /root/usr/sbin/update-initramfs
-    else
-        cat > /root/usr/sbin/update-initramfs <<EOF
-#! /bin/sh
-echo "update-initramfs is disabled since running on read-only media"
-exit 0
-EOF
-        chmod 755 /root/usr/sbin/update-initramfs
-    fi
-fi
+#force read only to protect from corruption as a result of premature removal
+#echo grep -q /root/cdrom /proc/self/mountinfo
+grep -q /root/cdrom /proc/self/mountinfo
+#echo "mount -o remount,ro $(grep /cdrom /proc/self/mountinfo | awk '{print $9}')"
+mount -o remount,ro $(grep /cdrom /proc/self/mountinfo | awk '{print $9}')
+grep -q /root/cdrom /proc/self/mountinfo
+
+# Running off a USB disk or other writable media.
+#if [ ! -e /root/usr/sbin/update-initramfs ] && \
+#   [ ! -L /root/usr/sbin/update-initramfs ]; then
+#    if [ -w /root/cdrom ] && \
+#       # rw is guaranteed to be first.
+#       grep -q ' /root/cdrom rw[, ]' /proc/self/mountinfo; then
+#
+#        ln -s /usr/share/casper/casper-update-initramfs \
+#            /root/usr/sbin/update-initramfs
+#    else
+#        cat > /root/usr/sbin/update-initramfs <<EOF
+##! /bin/sh
+#echo "update-initramfs is disabled since running on read-only media"
+#exit 0
+#EOF
+#        chmod 755 /root/usr/sbin/update-initramfs
+#    fi
+#fi
 
 log_end_msg
